pkg/controller/projectreference/projectreference_adapter.go

```diff
@@ -101,46 +102,81 @@ func NewReferenceAdapter(projectReference *gcpv1alpha1.ProjectReference, logger
 	}, nil
 }
 
-func (r *ReferenceAdapter) EnsureProjectClaimReady() (gcpv1alpha1.ClaimStatus, error) {
+func EnsureProjectClaimReady(r *ReferenceAdapter) (OperationResult, error) {
 	if r.ProjectReference.Status.State != gcpv1alpha1.ProjectReferenceStatusReady {
-		return r.ProjectClaim.Status.State, nil
+		return ContinueProcessing()
 	}
 
 	if r.ProjectReference.Status.State == gcpv1alpha1.ProjectReferenceStatusReady && r.ProjectClaim.Status.State == gcpv1alpha1.ClaimStatusReady {
-		return r.ProjectClaim.Status.State, nil
+		return StopProcessing()
 	}
 
 	azResult, err := r.ensureClaimAvailabilityZonesSet()
 	if err != nil {
-		return r.ProjectClaim.Status.State, operrors.Wrap(err, "error ensuring availability zones")
+		return RequeueWithError(operrors.Wrap(err, "error ensuring availability zones"))
 	}
 
 	idModified := r.ensureClaimProjectIDSet()
 
 	if azResult == ensureAzResultModified || idModified {
 		err := r.kubeClient.Update(context.TODO(), r.ProjectClaim)
 		if err != nil {
-			return r.ProjectClaim.Status.State, operrors.Wrap(err, "error updating ProjectClaim spec")
+			return RequeueWithError(operrors.Wrap(err, "error updating ProjectClaim spec"))
 		}
+		return StopProcessing()
 	}
 
 	if azResult == ensureAzResultNotReady {
-		return r.ProjectClaim.Status.State, nil
+		r.logger.V(int(logtypes.ProjectReference)).Info("Compute API not yet fully initialized. Retrying in 30 seconds.")
+		return RequeueAfter(30*time.Second, nil)
```

Comment:
gofmt doesn't seem to allow that space

Reply:
I don't like the additional line this would add everywhere. I think it's obvious from the function call with or without space. I'd do RequeueWithLongDelay(nil). But I also like that we actually show the time we delay.

Reply:
agreed, let's leave it at that.
maybe extract to constants?
`shortDelay` and `longDelay`?

Reply:
nvm. let's just leave at that and worry about it in another day